modules/caddyhttp/reverseproxy/reverseproxy.go

```diff
@@ -1348,6 +1356,74 @@ func statusError(err error) error {
 	return caddyhttp.Error(statusCode, err)
 }
 
+// delayClientGoneContext is a special context.Context type
+// intended for use when doing a RoundTrip where you don't
+// want a client disconnection to cancel the request during
+// the roundtrip. We delay the context.Done() channel until
+// the round trip is done, or until the parent context is
+// done, whichever comes first.
+type delayClientDoneContext struct {
```

Comment:
I think instead of introducing a new type, we can just use `context.WithCancel` with `h.ctx` and `context.AfterFunc` with the request context and cancel the context introduced before to achieve the same effect. Not only does this save many lines of code, it will also avoid the `Done` returning different channel inconsistency with stdlib. What do you think?

```
		ctx, cancel := context.WithCancel(h.ctx)
		defer cancel()
		stop := context.AfterFunc(req.Context(), cancel)
		defer stop()
```